File: DotnetAsm.Roslyn/SemanticHighlight/Models/SemanticHighlightSpan.cs
namespace DotnetAsm.Roslyn.SemanticHighlight.Models;

public class SemanticHighlightSpan
{
    public required int StartLine { get; init; }
    public required int StartColumn { get; init; }
    public required int EndLine { get; init; }
    public required int EndColumn { get; init; }
    public required SemanticHighlightClassification ClassificationType { get; init; }
}

File: DotnetAsm.Frontend/src/models/Roslyn/SemanticHighlight/SemanticHighlightSpan.ts
import SemanticHighlightClassification from "./SemanticHighlightClassification";

type SemanticHighlightSpan = {
  startLine: number;
  startColumn: number;
  endLine: number;
  endColumn: number;
  classificationType: SemanticHighlightClassification;
};

export default SemanticHighlightSpan;
